packages/test-utils/src/matches.js

```diff
@@ -56,13 +56,7 @@ export function matches(node, selector) {
     return element && element.matches && element.matches(selector.value)
   }
 
-  const isFunctionalSelector = isConstructor(selector.value)
-    ? selector.value.options.functional
-    : selector.value.functional
-
-  const componentInstance = isFunctionalSelector
-    ? node[FUNCTIONAL_OPTIONS]
-    : node.child
+  const componentInstance = node[FUNCTIONAL_OPTIONS] || node.child
```

Comment:
Looks like the line you removed was for back compat. You might need to skip the failing test. Are you able to update the PR and fix the failing test?

Reply:
Should I just skip this test for vue < 2.3?

Reply:
Because I find no way to throw exception in this case

Reply:
Sure, that's probably fine, go for it. I think most people will be on at least 2.5 by now.